__tests__/issue_9_spec.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { fireEvent, render } from '@testing-library/react'
+import { atom, useAtom } from 'jotai'
+import { atomsWithTanstackQuery } from '../src/index'
+
+describe('issue #9', () => {
+  it('status should change', async () => {
+    const idAtom = atom(undefined as number | undefined)
+
+    const [, statusAtom] = atomsWithTanstackQuery((get) => ({
+      queryKey: ['users', get(idAtom)],
+      queryFn: async ({ queryKey: [, id] }) => {
+        await new Promise((resolve) => setTimeout(resolve, 100))
```

Comment:
IMHO better to resolve this explicitly rather than relying on timing. This test might be flaky on busy machine (e.g. if process suddenly get preempted for >100ms between lines 56 & 57.

Reply:
Yeah, I always felt that. Good point. I can try.

Reply:
f8cbdaa
Do you have a better idea?
We did so many setTimeout hack in https://github.com/pmndrs/jotai tests, and I wanted someone for help.

Reply:
> [f8cbdaa](https://github.com/jotai-labs/jotai-tanstack-query/commit/f8cbdaa3d98f10fe8830d9f909bf8ee5f9ade6f1) Do you have a better idea?

This is probably as good as it gets. I didn't find any popular testing lib with promise mocks.
If this is used many times then probably some kind of helper could be handy.

```ts
function mockPromise<T = void>() {
    let resolve: (value: T) => void;
    let promise = new Promise<T>((r) => resolve = r);
    return Object.assign(promise, {resolve: resolve!});
}

const p = mockPromise();
p.then(() => console.log("hello"));
p.resolve();
```